### DISCUSSION TITLE
Requesting locate route, 'nodes' key is empty.

### DISCUSSION THREAD

**Discussion by Sh4ki24** [gis-ops/docker-valhalla#140] [Q&A]
Hello dear github developers who are working here 👋🏻 I need your help.

Valhalla is created with this repo and is up and running, but i can't get data related to the type of street. based on the [valhalla:locate](https://valhalla.github.io/valhalla/api/locate/api-reference/) document, when 'verbose' is true I should have access to nodes->type, but I receive nothing. I tried to configure the docker-compose provided in the repository but it didn't work, I couldn't make it. can you help me with that?

``` 
http://localhost:8002/locate?json={"verbose":true,"locations":[{"lat":35.748741,"lon":51.254665}, {"lat":35.748741,"lon":51.264665}],"costing":"bicycle","costing_options":{"bicycle":{"bicycle_type":"road"}},"directions_options":{"units":"miles"},"id":"12abc3afe23984fe"}

```

```
http://localhost:8002/locate?json={"verbose":true,"locations":[{"lat":35.748741,"lon":51.254665}, {"lat":35.748741,"lon":51.264665}]}
```

<img width="776" alt="image" src="https://github.com/gis-ops/docker-valhalla/assets/50788815/5964fddc-23cc-44a5-9670-473cb0592eeb">

<img width="464" alt="image" src="https://github.com/gis-ops/docker-valhalla/assets/50788815/6766ab03-d2ee-413f-b3c3-af68068e446e">

**Comment by nilsnolde** [comment]
well, only if there's any nodes nearby. nodes in routing lingo are what most people would call intersections. if you click in the middle of an edge you won't get any nodes by default.

**Reply by Sh4ki24** [reply]
Thanks for the response. The lat,lng coordinates I provided correspond to a point on a highway and I was expecting it to return something like "highway".
 Is there any other way to receive the type of a route?
could this be due to the country I'm querying?

**Reply by nilsnolde** [reply]
that has little to do with "nodes" though. you want information about an "edge", i.e. smth that has to do with OSM way. that's of course in "edges", not in "nodes". just look for "classification" in the returned JSON. it's also pretty clearly given as example in https://valhalla.github.io/valhalla/api/locate/api-reference/#outputs-of-a-locate-request

**Reply by Sh4ki24** [reply]
To clarify my subject, I need data of nodes and edges both. As you can see in the provided picture, some points are located at intersections, and some are on roads. That's why I was looking for nodes data as well.

![nodes.jpg](https://github.com/gis-ops/docker-valhalla/assets/50788815/60b1fba9-d6a1-401b-baf6-2da7467adf11)
